src/Microsoft.AspNet.Hosting/HostingEngine.cs

```diff
@@ -56,14 +59,20 @@ public virtual IDisposable Start()
 
             var applicationDelegate = BuildApplicationDelegate();
 
-            var _contextFactory = _applicationServices.GetRequiredService<IHttpContextFactory>();
-            var _contextAccessor = _applicationServices.GetRequiredService<IHttpContextAccessor>();
+            var logger = _applicationServices.GetRequiredService<ILogger<HostingEngine>>();
+            var contextFactory = _applicationServices.GetRequiredService<IHttpContextFactory>();
+            var contextAccessor = _applicationServices.GetRequiredService<IHttpContextAccessor>();
             var server = _serverFactory.Start(_serverInstance,
                 features =>
                 {
-                    var httpContext = _contextFactory.CreateHttpContext(features);
-                    _contextAccessor.HttpContext = httpContext;
-                    return applicationDelegate(httpContext);
+                    var httpContext = contextFactory.CreateHttpContext(features);
+                    var requestIdentifier = GetRequestIdentifier(httpContext);
+
+                    using (logger.BeginScope("Request Id: {RequestId}", requestIdentifier))
+                    {
+                        contextAccessor.HttpContext = httpContext;
+                        return applicationDelegate(httpContext);
```

Comment:
:+1: I suppose you are referring to the scope(async local) availability...I initially had async-await, but wasn't sure if it was necessary...added it now...